### PR TITLE
chore(Carta Giovani Nazionale): [IOACGN-32] Handles new ProductCategories for CGN

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR implements new categories on CGN

This PR depends on pagopa/io-backend#866 and pagopa/io-dev-api-server#127

As an improvement this PR adds:
- pull to refresh on Merchants list
- minor fixes on discount code visualization

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -2573,15 +2573,16 @@ bonus:
...
-        travel: travels, transportation and mobility
-        food: restaurant and cooking
-        hotel: hotel
-        service: services
-        book: instruction
-        shopping: shopping
-        sport: sport
+        cultureAndEntertainment: culture and entertainment
         health: health and wellness
+        learning: education and training
```
Comment (CrisTofani) [inline locales/en/index.yml:2578]:
```suggestion
        learning: education and learning
```

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "io_content_specs": "https://raw.githubusercontent.com/pagopa/io-services-metadata/1.0.3/definitions.yml",
   "io_bonus_vacanze_specs": "https://raw.githubusercontent.com/pagopa/io-backend/v7.31.0-RELEASE/api_bonus.yaml",
   "io_cgn_specs": "https://raw.githubusercontent.com/pagopa/io-backend/v7.32.0-RELEASE/api_cgn.yaml",
-  "io_cgn_merchants_specs": "https://raw.githubusercontent.com/pagopa/io-backend/v7.30.3-RELEASE/api_cgn_operator_search.yaml",
+  "io_cgn_merchants_specs": "https://raw.githubusercontent.com/pagopa/io-backend/v7.32.1-RELEASE/api_cgn_operator_search.yaml",
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply package.json:9]: ce323e2 done!

File: ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx [hunk 3]
```diff
@@ -57,19 +57,51 @@ const CgnMerchantDiscountItem: React.FunctionComponent<Props> = ({
...
+                          {
+                            count: discount.productCategories.length - 1
+                          }
+                        ).toLocaleUpperCase()}
+                      </H5>
+                    </>
+                  )
+                )
+              : discount.productCategories.map(category => (
+                  <>
```
Comment (fabriziofff) [inline ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx:87]:
```suggestion
              : discount.productCategories.map((category, i) => (
                  <View key={i}>
```
Like the previous comment, in order to have unique keys for react children.

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx [hunk 4]
```diff
@@ -55,22 +54,32 @@ export const CgnDiscountDetail: React.FunctionComponent<Props> = ({
...
-          </H5>
-        </View>
-      ))
-    )}
+  <View style={[styles.container, IOStyles.flex]} testID={"discount-detail"}>
+    <View style={[styles.row, IOStyles.flex, { flexWrap: "wrap" }]}>
+      {discount.productCategories.map(categoryKey =>
+        getCategorySpecs(categoryKey).fold(undefined, c => (
+          <View
+            style={[
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx:63]: fixed for both comments and actually noticed that the same warning was present on `CgnMerchantListItem` for the same reason, just fixed it

File: ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx [hunk 5]
```diff
@@ -55,22 +54,32 @@ export const CgnDiscountDetail: React.FunctionComponent<Props> = ({
...
+          >
+            {c.icon({
+              height: CATEGORY_ICON_SIZE,
+              width: CATEGORY_ICON_SIZE,
+              fill: IOColors.bluegrey
+            })}
+            <View hspacer small />
+            <H5 weight={"SemiBold"} color={"bluegrey"} testID={"category-name"}>
+              {I18n.t(c.nameKey).toLocaleUpperCase()}
+            </H5>
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx:80]: testing with real data didn't showed up this case with 3 long named categories, thank for catching it. Latest commit should add some space

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**